rust-src/wallet_library/src/statement.rs

```diff
@@ -0,0 +1,274 @@
+use concordium_base::{
+    curve_arithmetic::Curve,
+    id::{id_proof_types::AtomicStatement, types::*},
+    web3id::{CredentialStatement, Request},
+};
+
+const IDENTITY_RANGE_TAGS: &[AttributeTag] = &[AttributeTag(3), AttributeTag(9), AttributeTag(10)]; // dob, idDocIssuedAt, idDocExpiresAt
+const IDENTITY_SET_TAGS: &[AttributeTag] = &[
+    AttributeTag(4),
+    AttributeTag(5),
+    AttributeTag(6),
+    AttributeTag(8),
+]; // Country of residence, Nationality, IdDocType, IdDocIssuer
+
+fn is_iso8601(date: &str) -> bool {
+    println!("{}", date);
+    chrono::NaiveDate::parse_from_str(date, "%Y%m%d").is_ok()
+}
+
+fn is_iso3166_alpha_2(code: &str) -> bool { rust_iso3166::from_alpha2(code).is_some() }
+
+fn is_iso3166_2(code: &str) -> bool { rust_iso3166::iso3166_2::from_code(code).is_some() }
+
+pub trait WalletCheck {
+    fn satisfies_wallet_restrictions(&self) -> bool;
+}
```

Comment:
Please document what this is for.

Reply:
Also why is this called "WalletCheck" when it's implemented for requests and statements.

It seems there is no "wallet" parameter anywhere. And how will we handle that some wallets, such as the browser wallet, have looser restrictions?

Reply:
I have renamed it as suggested (And split it into two traits)
(And added a WalletConfig parameter, as suggested)